File: src/LC1170.java
import java.lang.reflect.Array;
import java.util.Arrays;

class Solution {
    public int[] numSmallerByFrequency(String[] queries, String[] words) {
        int[] lens = new int[words.length];
        int[] times;
        int j=0;
        int temp=0;
        for(int i=0;i<words.length;i++)
        {
            times = new int[26];

            for(j=0;j<words[i].length();j++)
            {
                times[words[i].charAt(j)-'a']++;
            }

            for(j=0;j<26;j++)
            {
                if(times[j]!=0)
                {
                    lens[i]=times[j];
                   break;
                }
            }
        }
        Arrays.sort(lens);
        int[] results = new int[queries.length];
        int tempSum;
        for(int i=0;i<queries.length;i++)
        {
            tempSum=0;
            times = new int[26];

            for(j=0;j<queries[i].length();j++)
            {
                times[queries[i].charAt(j)-'a']++;
            }

            for(j=0;j<26;j++)
            {
                if(times[j]!=0)
                {
                    break;
                }
            }
            for(int k=0;k<lens.length;k++)
            {
                if(times[j]<lens[k])
                    tempSum++;
            }
            results[i] = tempSum;
        }
        return results;
    }
}

File: src/LC1170.cpp
#include "vector"
#include "string"

using namespace std;

class Solution {
public:
    vector<int> numSmallerByFrequency(vector<string>& queries, vector<string>& words) {
        vector<int> counter(2001);
        vector<int> subfix(2001);

        for (auto& w : words) {
            int count = f(w);
            counter[count]++;
        }

        for (int i = static_cast<int>(counter.size()) - 2; i >= 0; --i) {
            subfix[i] = counter[i + 1] +subfix[i + 1] ;
        }

        vector<int> res(queries.size());

        for (auto i = 0; i < queries.size(); i++) {
            int count = f(queries[i]);
            res[i] = subfix[count];
        }
        return res;
    }

    int f(const string& w) {
        char c = 'z';
        int count = 0;
        for (auto& ch : w) {
            if (ch < c) {
                c = ch;
                count = 1;
            } else if (ch == c) {
                count++;
            }
        }
        return count;
    }
};